### PR TITLE
526 reporting

### PR DESCRIPTION
**Pull request by kreek**
## Description of change
Adds classes needed to track and report on the status of a 526 submission workflow.

## Testing done
Unit tests

## Testing planned
Staging integration tests

#### Unique to this PR
- [ ] Adds a 526 workflow steps tracker class
- [ ] Adds a 526 workflow report class
- [ ] Adds needed migrations for the `disability_compensation_submissions` table
- [ ] Refactors 526 submit job
- [ ] Extracts 526 metric logic to its own class

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: app/workers/evss/disability_compensation_form/workflow_steps.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    class WorkflowSteps
+      def set_has_uploads(id)
+        submission(id).update(has_uploads: true)
+      end
+
+      def set_has_form_4142(id)
+        submission(id).update(has_form_4142: true)
+      end
+
+      def uploads_success_handler(_status, options)
```
Comment (abhijitvk) [inline app/workers/evss/disability_compensation_form/workflow_steps.rb]: @kreek : For batch failures, don't you need another method to update the column with false value? Likely set false along with a call to WorkflowSteps.new.set_has_uploads and then later on success it updates to true?

File: db/migrate/20180924231236_add_disability_compensation_worklow_fields.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class AddDisabilityCompensationWorklowFields < ActiveRecord::Migration
+  def change
+    add_column :disability_compensation_submissions, :has_uploads, :boolean
+    add_column :disability_compensation_submissions, :uploads_success, :boolean
+    add_column :disability_compensation_submissions, :has_form_4142, :boolean
```
Comment (abhijitvk) [inline db/migrate/20180924231236_add_disability_compensation_worklow_fields.rb]: @kreek : In the future for any new document type / forms, are you thinking we would add new columns to the table, for example upcoming forms 781, 8940. 

Just some initial comment for extensibility. Have you given any thought of capturing "doc_type", "form_id" and "upload_status" columns in a new table joined by possibly "va526ez_submit_transaction_id " and "disability_compensation_id" columns?

File: spec/jobs/evss/disability_compensation_form/workflow_report_spec.rb [hunk 3]
```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe EVSS::DisabilityCompensationForm::WorkflowReport do
+  let(:user) { build(:user, :loa3) }
+  let(:saved_claim) { FactoryBot.create(:va526ez) }
+  let(:async_transaction) { FactoryBot.create(:va526ez_submit_transaction) }
```
Comment (SMLuthi) [inline spec/jobs/evss/disability_compensation_form/workflow_report_spec.rb]: I dont think this assignment is necessary for the tests.

File: app/workers/evss/disability_compensation_form/workflow_steps.rb (hunk 1 above)
Comment (kreek) [reply app/workers/evss/disability_compensation_form/workflow_steps.rb]: The default value for the success columns is false :)

File: db/migrate/20180924231236_add_disability_compensation_worklow_fields.rb (hunk 2 above)
Comment (kreek) [reply db/migrate/20180924231236_add_disability_compensation_worklow_fields.rb]: 👍Good call. I thought 781 and 8940 were the extent of the ancillary items. Checked with Dawn and there's a least a dozen more ancillary forms 😬. It does mean uploads would then be the outlier with a different pattern. Looking at the fields again we may not actually need the `has_uploads` and `has_form_4142` etc. fields as they could be inferred through the form data in the referenced saved claim table.

File: app/workers/evss/disability_compensation_form/workflow_steps.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module EVSS
+  module DisabilityCompensationForm
+    class WorkflowSteps
+      def set_has_uploads(id)
+        submission(id).update(has_uploads: true)
+      end
+
+      def set_has_form_4142(id)
```
[the remark this answers is not in the document]
Comment (kreek) [reply app/workers/evss/disability_compensation_form/workflow_steps.rb]: It's not called yet, I was going to add to add it in a later PR :)

**Comment by kreek** [comment]
Closing this while we redesign how 526 reporting will work. Opened another PR that includes just the refactoring work from this one: https://github.com/department-of-veterans-affairs/vets-api/pull/2314.

**Pull request closed without merging**